### ISSUE TITLE
Support UTF-8 in localpart of an email address for mail submission (SMTPUTF8)

### ISSUE THREAD

**Issue by rcubetrac** [roundcube/roundcubemail#5120]
_Reported by mmartinec on 26 Sep 2014 16:00 UTC as Trac ticket #1490082_

Now that mailers at gmail.com, as well as the postfix mailer (version 2.12), support the "SMTP Extension for Internationalized Email (SMTPUTF8)" (RFC 6531) in SMTP and LMTP mail submission, it would be really nice to be able to send mail from roundcube to a recipient with non-ASCII UTF-8 localpart in his e-mail address (an EAI address), provided that a mailer (MSA) announces support for SMTPUTF8.

Currently (roundcube 1.0.2) does not allow to specify such email address (e.g. Nez@example.com), even if a mailer does support SMTPUTF8.

Luckily at least the internationalized domain names are already handled well by roundcube for a couple of years now, i.e. an UTF-8 domain name(IDN) is transformed by roundcube into an ASCII-compatible encoding (ACE), and the other way around when displaying such domain name - kudos for doing it well!!! --- Although this conversion step to A-labels may be omitted when submitting mail to a mail submission agent announcing the SMTPUTF8 support.

The relevant RFCs are:
- RFC 6530 - Overview and Framework for Internationalized Email
- RFC 6531 - SMTP Extension for Internationalized Email (SMTPUTF8)
- RFC 6532 - Internationalized Email Headers
- RFC 6533 - Internationalized Delivery Status Notifications

So in summary, it should be possible to send mail from roundcube to jran@blbrsyltety.gulbrandsen.priv.no (not just to postmaster@blbrsyltety.gulbrandsen.priv.no) if MSA announces support for SMTPUTF8. (btw, the first address above is an autoresponder and may be used for testing mail internationalization)

(Independent but related: for IMAP & POP3 this is governed by RFC 6855, RFC 6856, RFC 6857, RFC 6858, although accessing such mail is independent from sending it, so this is a topic of some other ticket)

_Keywords: SMTPUTF8 EAI UTF-8 internationalization_
_Migrated-From: http://trac.roundcube.net/ticket/1490082_

**Comment by vbertola** [comment]
Hi, is there any chance that this will be implemented? EAI is gaining adoption, in India the government is giving several millions of fully internationalized email addresses (UTF-8@IDN) to public officers, what if I have to exchange email with one of them? Even in Europe, all languages except English have extended characters that are progressively coming into use in URLs and email addresses and require IDN/EAI support. So please put this on your roadmap...

**Comment by alecpl** [comment]
There is a PR #5256 to support this. It requires review and testing. I'll see if I can do this for 1.4.

**Comment by alecpl** [comment]
Fixed.
